cfgov/searchgov/views.py

```diff
@@ -0,0 +1,106 @@
+import math
+
+from django.conf import settings
+
+import requests
+
+from core.feature_flags import environment_is
+from core.views import TranslatedTemplateView
+
+from .forms import SearchForm
+
+
+API_ENDPOINT = "https://api.gsa.gov/technology/searchgov/v2/results/i14y?affiliate={}&access_key={}&limit={}&offset={}&query={}"
+RESULTS_PER_PAGE = 20
+
+
+def get_affiliate(context):
+    """Given a request, return the appropriate affiliate for Search.gov.
+    Our default affiliate code is "cfpb". We have a separate Spanish-language
+    index named "cfpb_es". We then have two additional indexes, "cfpb_beta"
+    and "cfpb_beta_es", for use on beta.consumerfinance.gov.
+    """
+    affiliate = "cfpb"
+
+    if environment_is("beta"):
+        affiliate += "_beta"
+
+    if context.get("current_language") == "es":
+        affiliate += "_es"
+
+    return affiliate
+
+
+def get_api_key(context):
+    if context.get("current_language") == "es":
+        return settings.SEARCHGOV_ES_API_KEY
+    return settings.SEARCHGOV_API_KEY
+
+
+class SearchView(TranslatedTemplateView):
```

Comment:
Pony request: it'd be nice if this view had a JSON version for easier testing of search results. One might consider using [this TCCP approach](https://github.com/cfpb/consumerfinance.gov/blob/9c4dcedcda084a060c506f25102e41afc0deb589/cfgov/tccp/views.py#L79), to implement that.

Reply:
oooh, that's a nice idea. I'll create a follow-up issue.